java/http/java-http-consumer/src/main/java/HttpConsumer.java

```diff
@@ -171,7 +179,9 @@ public void poll() {
                     log.info("Record {}", record);
                     this.messageReceived++;
                 }
-
+                if (records.size() > 0 && this.manualCommit) {
+                    this.commitMessages();
+                }
```

Comment:
you are committing messages if manualCommit (or !enableAutoCommit, see my previous comment), but still the underlying Kafka consumer on the bridge is configured by auto committing because it's the default for Kafka (see https://kafka.apache.org/documentation/#consumerconfigs_enable.auto.commit).
I don't want to spoiler everything but remember my comment ...

> If we do really want to have manual commit, we should create the HTTP consumer with auto commit disabled or we should make it configurable (which is something we have in the KafkaConsumerExample).

What I see you are adding the `KAFKA_` related stuff from the Kafka examples but they are not used at all to configure the corresponding Kafka consumer via the HTTP consumer creation. You are using the properties just for getting the auto commit configuration.
Using those properties is simple for a Kafka based example because the creation of the Kafka consumer just gets ... a `Properties` instance, which is not true for an HTTP consumer creation.
See https://strimzi.io/docs/bridge/latest/#proc-creating-kafka-bridge-consumer-bridge

With your current code, what happens if the user specifies a KAFKA_AUTO_OFFSET_RESET for example? Are you just ignoring it? Should it configure the corresponding HTTP consumer?

Reply:
This seems to be resolved now by creating a new JSON with the configs payload. Hashmap is used though ObjectNode may be a better route.